cluster/replication/consumer_ops_cache.go

```diff
@@ -11,15 +11,24 @@
 
 package replication
 
-import "sync"
+import (
+	"context"
+	"sync"
+)
 
 type OpsCache struct {
+	// canceleds is a map of opId to an empty struct
+	cancelleds sync.Map
+	// cancels is a map of opId to a cancel function
+	cancels sync.Map
+	// ops is a map of opId to an empty struct
 	ops sync.Map
 }
 
 func NewOpsCache() *OpsCache {
 	return &OpsCache{
-		ops: sync.Map{},
+		cancels: sync.Map{},
+		ops:     sync.Map{},
```

Comment:
nit, can we either explicitly init all of the sync map's on OpsCache or none of them? just avoids people (eg me reading this pr) trying to figure out if there is some reason we don't init cancelleds

Reply:
Yes, that's a mistake on my part for not adding!

Reply:
I have instead completely removed the `cancelleds` field from the cache since it is not necessary to achieve what I initially required from it. I had [this](https://go.dev/blog/codelab-share) in mind when implementing it originally and realised upon fresh eyes that there is a simpler way to achieve it in the logic flow. Namely that the context cancelled error can be used to trigger the updated cancelled state